pkg/metrics/ingest.go

```diff
@@ -41,12 +43,27 @@ type MQTTPayload map[string]interface{}
 
 func (i *Ingest) store(deviceID string, rawMetrics MQTTPayload) error {
 	var mc MetricCollection
+
 	for metricName, value := range rawMetrics {
 		if cfg, found := i.validMetrics[metricName]; found {
-
-			floatValue, ok := value.(float64)
-			if !ok {
-				return fmt.Errorf("got data with unexpectd type: %T ('%s')  but wanted float64", value, value)
+			var floatValue float64
+			var isFloat bool
+			var err error
+			floatValue, isFloat = value.(float64)
+			if !isFloat {
+				stringvalue, isString := value.(string)
+				if isString {
```

Comment:
Can change the order of the branches? If we change this to:
```
if !isString || ! validNumber.MatchString(stringvalue){
   return fmt.Errorf("got data with unexpectd type: %T ('%s')", value, value) 
}
```
The indentation level is reduced.